src/iop/demosaic.c

```diff
@@ -1502,6 +1502,15 @@ void modify_roi_in(struct dt_iop_module_t *self, struct dt_dev_pixelpipe_iop_t *
   // this op is disabled for preview pipe/filters == 0
 
   *roi_in = *roi_out;
+
+  if ((piece->pipe->type == DT_DEV_PIXELPIPE_PREVIEW) &&
+      self->dev->image_storage.buf_dsc.filters)
+  {
+    // This is a mosaiced MIP_F, and was given a larger buffer as
+    // 1-channel. Shrink to proper size now that it is 4-channel.
+    roi_in->scale /= 2.0f;
```

Comment:
I realize this bit is utterly wrong, and doesn't in fact halve the image size as hoped for. Still looking for the right way to do this.